pkg/controllers/user/ingresshostgen/ingresshostgen.go

```diff
@@ -60,16 +62,42 @@ func (i *IngressHostGen) sync(key string, obj *v1beta1.Ingress) (runtime.Object,
 		}
 	}
 
-	if !changed {
+	ingressState := ingress.GetIngressState(obj)
+	requiredKeysUpdate := false
```

Comment:
Sorry for the late reply.
1. We can also not generate the state key in the creation via api because we don't really need the ingress service if the `xip.io` is not get generated.
2. The separate controller is only for updating the state key generated with `xip.io` as host. The ingressHostGen controller logic will be updating the ingress rule with `xip.io` and set the state key for it. I think it will be easier to understand without the upgrading logic in it.
3. Need to make an example for this one. But We can go on fixing this issue without it.